app/client/components/productPage.tsx

```diff
@@ -152,26 +156,51 @@ const getPaymentPart = (
   productDetail: ProductDetail,
   productType: ProductType
 ) => {
-  if (productDetail.isPaidTier) {
+  const mainPlan = getMainPlan(productDetail.subscription);
+  const futurePlan = getFuturePlanIfStartsBeforeXDaysFromToday(
+    productDetail.subscription
+  );
+  if (isPaidSubscriptionPlan(mainPlan)) {
+    const mainPlanInterval = augmentInterval(mainPlan.interval);
     return (
       <>
-        {productDetail.subscription.nextPaymentDate && (
-          <ProductDetailRow
-            label={"Next payment date"}
-            data={formatDate(productDetail.subscription.nextPaymentDate)}
-          />
-        )}
+        {productDetail.subscription.nextPaymentDate &&
+          !productDetail.alertText && (
+            <ProductDetailRow
+              label={"Next payment date"}
+              data={formatDate(productDetail.subscription.nextPaymentDate)}
+            />
+          )}
+        {}
         <ProductDetailRow
           label={
-            productDetail.subscription.plan.interval.charAt(0).toUpperCase() +
-            productDetail.subscription.plan.interval.substr(1) +
-            "ly payment"
+            mainPlanInterval.charAt(0).toUpperCase() +
+            mainPlanInterval.substr(1) +
+            " payment"
           }
           data={
-            <UpdatableAmount
-              subscription={productDetail.subscription}
-              productType={productType}
-            />
+            <>
+              <UpdatableAmount
+                mainPlan={mainPlan}
+                subscriptionId={productDetail.subscription.subscriptionId}
+                productType={productType}
+              />
+              {futurePlan &&
+                futurePlan.productRatePlanId !== mainPlan.productRatePlanId &&
```

Comment:
Do we need the productRatePlanId comparison here? I think comparing the amounts would be sufficient?

Reply:
+1, I agree with Jacob’s earlier comment that we shouldn’t expose the productRatePlanId from Zuora.

Reply:
removed in https://github.com/guardian/manage-frontend/pull/196/commits/3dd81ad407ac94a95d6158ed5e428bb39e2365fc